File: Algorithms/p405_Convert_a_Number_to_Hexadecimal.cpp
class Solution {
public:
    string toHex(int num) {
        if(num!=0){
            unsigned int n = num;
            string res;
            while(n){
                int d = n&0xf;
                if(d>=0 && d<10){
                    res.insert(res.begin(),'0'+d);
                }else{
                    res.insert(res.begin(),'a'+d-10);
                }
                n >>= 4;
            }
            return res;
        }else{
            return "0";
        }
    }
};

File: Algorithms/p405_Convert_a_Number_to_Hexadecimal.py
class Solution(object):
    def toHex(self, num):
        """
        :type num: int
        :rtype: str
        """
        if num!=0:
            n = num if num>=0 else 0x100000000+num
            res = ''
            while n!=0:
                d = n&0xf
                if d>=0 and d<10:
                    res = chr(ord('0')+d)+res
                else:
                    res = chr(ord('a')+d-10)+res
                n >>= 4
            return res
        else:
            return '0'
